File: VB/FrameNavigationServiceSL/ViewModel/MainViewModel.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Xpf.Mvvm
Imports System.Windows.Input

Namespace FrameNavigationServiceSL.ViewModel
	Public Class MainViewModel
		Inherits ViewModelBase
		Private privateOnViewLoadedCommand As ICommand
		Public Property OnViewLoadedCommand() As ICommand
			Get
				Return privateOnViewLoadedCommand
			End Get
			Private Set(ByVal value As ICommand)
				privateOnViewLoadedCommand = value
			End Set
		End Property
		Public Sub New()
			OnViewLoadedCommand = New DelegateCommand(AddressOf OnNavigateDetailsCommandExecute)
		End Sub
		Private Sub OnNavigateDetailsCommandExecute()
			'Navigate to the HomeView.
			ServiceContainer.GetService(Of INavigationService)().Navigate("HomeView", Nothing, Me)
		End Sub
	End Class
End Namespace

File: CS/FrameNavigationServiceSL/FrameNavigationServiceSL/ViewModel/MainViewModel.cs
using DevExpress.Xpf.Mvvm;
using System.Windows.Input;

namespace FrameNavigationServiceSL.ViewModel {
    public class MainViewModel : ViewModelBase {
        public ICommand OnViewLoadedCommand { get; private set; }
        public MainViewModel() {
            OnViewLoadedCommand = new DelegateCommand(OnNavigateDetailsCommandExecute);
        }
        void OnNavigateDetailsCommandExecute() {
            //Navigate to the HomeView.
            ServiceContainer.GetService<INavigationService>().Navigate("HomeView", null, this);
        }
    }
}